bt-116: Create sign up talent page

Simpsonkk opened the pull request:
I also added 'stepContent' prop in which we will put our components like profile, BSA badges, skills, etc.
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/92392661/fa57e939-a0d2-4436-ab9a-e51313de1a76)
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/92392661/5e6a4747-a8c4-43e5-bc64-02c9aec38a85)

Simpsonkk committed:
bt-116: + sign up talent page

Simpsonkk committed:
Merge branch 'development' into task/bt-116-create-talent-sign-up-layout

Simpsonkk committed:
bt-116: * sidebar styles.module.scss

nikita-remeslov reviewed (changes requested):
Don't see any routing

Also, you should navigate to steps on click Next/Prev

Simpsonkk committed:
bt-116: + step navigation and tabs payload

Simpsonkk committed:
Merge branch 'task/bt-116-create-talent-sign-up-layout' of github.com:BinaryStudioAcademy/bsa-2023-bsa-talents into task/bt-116-create-talent-sign-up-layout

Simpsonkk commented:
I removed prop stepContent in sign-up-talent page because when the user clicks on the next button we will give him the next tab. 
I made it and added some mock data for testing. It will be also easier to replace this mock data with real tab components, I left comments around mock data in the code.
I added a step nav when the user clicks on the back/next buttons ( you can see the screenshot ).

TODO: Add a special layout for step 5.
![image_2023-08-29_19-22-30](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/92392661/6f8c8861-3f22-4e54-87e3-35990f40eaa9)

Simpsonkk committed:
bt-116: + mobile markup

vladjsp reviewed (approved):
For me it works great on my local machine. The only things I noticed it's folder structure inside frontend/auth/components. It seems that 'step' and 'step-content' are not general (main) components so maybe it's better to put these two components inside 'sign-up-content' since they only used here or move them into common/components if they may appear elsewhere in the project.

ptarho commented:
It works well on my PC too. But in figma on last step we display only preview block without steps side block and "Let`s get started text". On last screen next and back buttons changes on "save without publishing" and "publish now" - but I don`t know if it`s a part of this task. Also, currently when I click on next after the last step app crashes.

Simpsonkk commented:
> It works well on my PC too. But in figma on last step we display only preview block without steps side block and "Let`s get started text". On last screen next and back buttons changes on "save without publishing" and "publish now" - but I don`t know if it`s a part of this task. Also, currently when I click on next after the last step app crashes.

You are right. I wrote a note to myself - TODO: Add a special layout for step 5. I`m doing it now.

Simpsonkk commented:
> For me it works great on my local machine. The only things I noticed it's folder structure inside frontend/auth/components. It seems that 'step' and 'step-content' are not general (main) components so maybe it's better to put these two components inside 'sign-up-content' since they only used here or move them into common/components if they may appear elsewhere in the project.

I agree with moving these files to sign-up-content folder

Simpsonkk committed:
bt-116: + step 5 layout

nikita-remeslov commented on frontend/src/bundles/common/components/sidebar/sidebar.tsx:
merge develop, sidebar already in the development branch

Simpsonkk replied:
@nikita-remeslov 
For now we don't have pages which will move you to sign-up talent page therefore it's confusing what I should add. Ok, I added a new route on our current root page to give you access to the sign-up talent page and also added a header and sidebar in app.tsx because they will be used on almost every page after that we should add the condition in app.tsx 'if header and sidebar don't need on the page - hide them'.

nikita-remeslov replied:
@Simpsonkk Well, how can I check your work? The fact that we do not have SignIn does not prevent you from adding the necessary routes.
In another way, these are just pages that are in the project and do not work in any way!

nikita-remeslov reviewed (changes requested):
> Don't see any routing
Also, you should navigate to steps on click Next/Prev

I still don't see routing, how can I access these pages

![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/78347072/00763e42-5c29-4bf0-b4fb-750f3403433c)

Simpsonkk committed:
Merge branch 'development' into task/bt-116-create-talent-sign-up-layout

Simpsonkk committed:
bt-116: + routing

nikita-remeslov commented on frontend/src/app/app.tsx:
I think we can remove almost everything because it was added for example

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-content/steps/styles.module.scss:
```diff
@@ -0,0 +1,103 @@
+        width: 18px;
+        height: 18px;
+        background-color: var(--st-tropaz);
+        background-image: url("../../../../../assets/img/check-mark.svg");
```
use alias

nikita-remeslov reviewed (changes requested):
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/78347072/8c341489-c03b-4e53-b8be-9c3b8bc5db8b)
Broken styles

I think you should create a layout where you combine header, sidebar, and content. Without it you will calc the width for each new page
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/78347072/d71b3f42-c47a-49de-a5e7-4cac95a06f77)

simple example - https://codepen.io/AaronMachado/pen/azpRZM

nikita-remeslov commented:
create a route for each step of onboarding and make navigation 
- talent/onboarding/step/:step

Simpsonkk committed:
bt-116: + page layout

Simpsonkk committed:
bt-116: + routing

Simpsonkk committed:
bt-116: * resolve conflicts

Simpsonkk committed:
bt-116: * resolve type link issue

Simpsonkk committed:
bt-116: * resolve new conflicts

Simpsonkk committed:
Merge branch 'development' into task/bt-116-create-talent-sign-up-layout

Simpsonkk commented:
for now after refreshing the page  - 'currentStep' resets (state in SignUpTalent) therefore step description, step number and step menu come back to initial values. I can add 'currentStep' number in the session or local storage and get it after refreshing the page in currentStep.

Simpsonkk committed:
bt-116: + aliases for ts files

Simpsonkk committed:
Merge branch 'task/bt-116-create-talent-sign-up-layout' of github.com:BinaryStudioAcademy/bsa-2023-bsa-talents into task/bt-116-create-talent-sign-up-layout

Simpsonkk committed:
bt-116: * proper page refresh

nikita-remeslov commented on frontend/package.json:
```diff
@@ -14,7 +14,8 @@
         "build": "tsc --build --clean && vite build",
-        "preview": "vite preview"
+        "preview": "vite preview",
+        "order:css": "stylelint \"src/**/*.scss\" --fix"
```
why do we need this command?

nikita-remeslov replied:
you can run command locally if you need
`npm run lint:css -- --fix`

Simpsonkk replied:
removed

nikita-remeslov commented on frontend/src/bundles/auth/components/sign-up-content/constants/constants.ts:
constant should be UPPER_CASA

Simpsonkk replied:
done

nikita-remeslov commented on frontend/src/services/step/step.service.ts:
why do we need this service? we will save step in the DB

Simpsonkk replied:
removed

nikita-remeslov reviewed (commented):
Also, it should be an `onboarding` bundle. Maybe better `talent-onbording`

Simpsonkk committed:
Merge branch 'development' into task/bt-116-create-talent-sign-up-layout

Simpsonkk committed:
bt-116: * move files in talent-onboarding folder

Simpsonkk committed:
Merge branch 'task/bt-116-create-talent-sign-up-layout' of github.com:BinaryStudioAcademy/bsa-2023-bsa-talents into task/bt-116-create-talent-sign-up-layout

nikita-remeslov commented on frontend/src/bundles/common/helpers/get-sign-up-talent-step-route/get-sign-up-talent-step-route.ts:
this is not common helper

nikita-remeslov replied:
you can add it to the propper bundle.
rename it to `get-step-route.ts`

nikita-remeslov commented on frontend/src/bundles/talent-onboarding/pages/sign-up-talent/sign-up-talent.tsx:
onboarding.tsx

Simpsonkk committed:
bt-116: * minor changes

Simpsonkk committed:
bt-116: * raname folder to onboarding

Simpsonkk committed:
bt-116: * remove extra enum

savinaDaria reviewed (commented):
looks good
